<launch>
    <!-- global arguments -->
    <arg name="odom_topic" default="/tb3/controller/odom" />
    <arg name="image_topic" default="/tb3/camera/image" />
    <arg name="save_path" default="~/MyFiles/Workspace/tr_data/" />

    <arg name="image_resize_width" default="100" />
    <arg name="image_resize_height" default="50" />
    <arg name="patch_size" default="(9,9)" />

    <arg name="distance_threshold" default="0.2" />
    <arg name="angle_threshold_deg" default="10.0" />

    <!-- global params -->
    <param name="image_resize_width" value="$(arg image_resize_width)" />
    <param name="image_resize_height" value="$(arg image_resize_height)" />
    <param name="patch_size" value="$(arg patch_size)" />

    <param name="wait_for_ready" value="false" />
    <param name="save_gt_data" value="true" />

    <!-- collecting -->
    <node pkg="teach_repeat" type="data_collect.py" name="data_collect" output="screen">
        <remap from="odom" to="$(arg odom_topic)" />
        <remap from="image" to="$(arg image_topic)" />

        <param name="distance_threshold" value="$(arg distance_threshold)" />
        <param name="angle_threshold_deg" value="$(arg angle_threshold_deg)" />
    </node>

    <!-- saving -->
    <node pkg="teach_repeat" type="data_save.py" name="data_save" output="screen">
        <param name="save_dir" value="$(arg save_path)" />
        <param name="timestamp_folder" value="true" />
    </node>

    <!-- teleop_keyboard -->
    <arg name="teleop_keyboard" default="true" />
    <node name="teleop_keyboard" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" if="$(arg teleop_keyboard)">
        <remap from="cmd_vel" to="/tb3/controller/cmd_vel" />
    </node>
</launch>
